lxd/device/tpm.go

```diff
@@ -217,34 +221,63 @@ func (d *tpm) startVM() (*deviceConfig.RunConfig, error) {
 		},
 	}
 
-	var fdFiles []*os.File
+	// Remove old socket if needed.
+	_ = os.Remove(socketPath)
 
-	// Ensure passed file is closed after swtpm start has returned.
-	defer func() {
-		for _, file := range fdFiles {
-			_ = file.Close()
-		}
-	}()
-
-	shortPath, err := util.ShortenedFilePath(socketPath, &fdFiles)
+	// Trickery to handle paths > 108 chars.
+	socketFileDir, err := os.Open(filepath.Dir(socketPath))
```

Comment:
While not a blocker, this comment was unclear to me at first. I realized the "trickery" only when I saw the comment of the `shortenedFilePath` function.

Reply:
thats a fair point indeed, will update, thanks.

Reply:
will merge this for now so we can check the tests are passing and will open a follow up PR